.github/workflows/deploy_release.yml

```diff
@@ -0,0 +1,77 @@
+#
+#  Licensed to the Apache Software Foundation (ASF) under one or more
+#  contributor license agreements. See the NOTICE file distributed with
+#  this work for additional information regarding copyright ownership.
+#  The ASF licenses this file to You under the Apache License, Version 2.0
+#  (the "License"); you may not use this file except in compliance with
+#  the License. You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+name: Deploy Release
+
+# This workflow is intended to be run after successfully validation of the release
+# staged by stage_release.  It will release (or drop) the staging repository.
+#
+#
+# It requires the following:
+# variables:
+# KROXYLICIOUS_SONATYPE_TOKEN_USERNAME - Sonatype Access User Token Username
+# secrets:
+# KROXYLICIOUS_SONATYPE_TOKEN_PASSWORD - Sonatype Access User Token Password
+# KROXYLICIOUS_RELEASE_TOKEN             - GitHub PAT wih content/createPullRequest permission for Kroxylicious repo.
+#
+
+on:
+  workflow_dispatch:
+    inputs:
+      next-state:
+        description: 'Desired next state for the staging repository'
+        required: true
+        type: choice
+        default: drop
+        options:
+          - drop
+          - release
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: 'Check out repository'
+        uses: actions/checkout@v4
+        with:
+          ref: main # transition-staging-repository-state.sh requires only the pom.xml, being on the release branch is not important
+          token: ${{ secrets.KROXYLICIOUS_RELEASE_TOKEN }}
+
+      - name: 'Cache Maven packages'
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+
+      - name: 'Set up Java'
+        uses: actions/setup-java@v3
+        with:
+          java-version: 21
+          distribution: temurin
+          server-id: ossrh
+          server-username: SONATYPE_TOKEN_USERNAME # env variable for Sonatype username
+          server-password: SONATYPE_TOKEN_PASSWORD # env variable for Sonatype password
+          overwrite-settings: true
+
+      - name: 'Transition snapshot repository to desired state'
+        env:
+          SONATYPE_TOKEN_USERNAME: ${{ vars.KROXYLICIOUS_SONATYPE_TOKEN_USERNAME }}
+          SONATYPE_TOKEN_PASSWORD: ${{ secrets.KROXYLICIOUS_SONATYPE_TOKEN_PASSWORD }}
+        run: ./scripts/transition-staging-repository-state.sh -s ${{ github.event.inputs.next-state }}
+
```

Comment:
This can be done with [`gh release edit`](https://cli.github.com/manual/gh_release_edit) (or [`gh release delete`](https://cli.github.com/manual/gh_release_delete) in the case of deletion). Should be as simple as
```shell
gh release edit "${RELEASE_TAG}" --draft=false
```
But given you need the release tag for both commands you may need to do something to the workflow to pass that around, or to scrape it out from the list of all releases.